### PR TITLE
Update config to current orb starter pack config [semver:skip]

### PR DESCRIPTION
**Pull request by dsayling**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [x] README has been updated, if necessary

### Motivation, issues
Update the CI

### Description
Update the config to use the orb starter kit orb-tools orb and setup the alpha branch flow

### CODE REVIEW

**Comment by felicianotech** [comment]
@dsayling Before CircleCI can run for this, the initial dev version, `circleci/python@dev:Alpha` needs to exists. This needs to be published from your local terminal. Let me know if you need help at all doing this.

**Comment by felicianotech** [comment]
@dsayling It looks like you're still actively pushing to this PR so I removed myself as a reviewer. Just add me back when this is ready and ping me if you'd like to talk about it.

**Comment by dsayling** [comment]
huzzah - it's finally 🟢

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,41 +1,86 @@
...
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circeci/python
+          context: orb-publishing
+          add-pr-comment: false
+          fail-if-semver-not-indicated: false
+          publish-version-tag: true
```
Comment (KyleTryon) [inline .circleci/config.yml:78]: This requires SSH fingerprints. Commented out below

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,41 +1,86 @@
...
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circeci/python
+          context: orb-publishing
+          add-pr-comment: false
+          fail-if-semver-not-indicated: false
```
Comment (KyleTryon) [inline .circleci/config.yml:77]: This needs to be true

File: .circleci/config.yml [hunk 3]
```diff
@@ -1,41 +1,86 @@
...
+      # publish a semver version of the orb. relies on
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
+          orb-name: circeci/python
+          context: orb-publishing
+          add-pr-comment: false
```
Comment (KyleTryon) [inline .circleci/config.yml:76]: This should be true

File: .circleci/config.yml [hunk 4]
```diff
@@ -1,41 +1,86 @@
...
+    when: << pipeline.parameters.run-integration-tests >>
+    jobs:
+      - pip-install-test
+      # publish a semver version of the orb. relies on
+      # the commit subject containing the text "[semver:patch|minor|major|skip]"
+      # as that will determine whether a patch, minor or major
+      # version will be published or if publishing should
+      # be skipped.
+      # e.g. [semver:patch] will cause a patch version to be published.
+      - orb-tools/dev-promote-prod-from-commit-subject:
```
Comment (KyleTryon) [inline .circleci/config.yml:73]: This should match the node orb nearly exactly, with different SSH fingerprints
https://github.com/CircleCI-Public/node-orb/blob/9079fe7ae040b459e80e0a3c096161d569f2d3a6/.circleci/config.yml#L193

File: .circleci/config.yml [hunk 5]
```diff
@@ -1,41 +1,86 @@
...
+          orb-name: circeci/python
+          context: orb-publishing
+          add-pr-comment: true
+          fail-if-semver-not-indicated: true
+          publish-version-tag: true
+          bot-token-variable: GH_BOT_USER_TOKEN
```
Comment (KyleTryon) [inline .circleci/config.yml:79]: Not the value in context.

Comment (KyleTryon) [reply .circleci/config.yml:79]: bot-token-variable: GHI_TOKEN
          bot-user: cpe-bot

File: .circleci/config.yml [hunk 6]
```diff
@@ -1,41 +1,86 @@
...
+          bot-user: GH-BOT
```
Comment (KyleTryon) [inline .circleci/config.yml:80]: Not the value in the context

**Pull request merged**